add ipython → jupyter migration doc

minrk opened the pull request:
cc @ellisonbg, @willingc

closes jupyter/notebook#419

ellisonbg commented on docs/source/migrating.rst:
```diff
@@ -0,0 +1,189 @@
+
+`The Big Split <https://blog.jupyter.org/2015/04/15/the-big-split/>`__
+has changed a few things, moving the various language-agnostict
+components of IPython under the Jupyter umbrella. This doc should help
```
"This document describes what has changed, and how you will need to modify your code and configuration."

Carreau commented on docs/source/migrating.rst:
```diff
@@ -0,0 +1,189 @@
+In all cases, ``JUPYTER_DATA_DIR`` can be used to specify this location
+explicitly. Data files installed system-wide (e.g. in
+``/usr/local/share/jupyter``) have not changed. Per-user installation of
+these has changed from going in .ipython to the above locations:
```
backtick for `.ipython`.

damianavila replied:
`:` at the end of the sentence?

ellisonbg commented on docs/source/migrating.rst:
```diff
@@ -0,0 +1,189 @@
+components of IPython under the Jupyter umbrella. This doc should help
+you find what's changed, and how to keep doing what you've been doing.
+
+Where have all my files gone?
```
While I know we are in a reactive mode right now, I think the document will be more useful if it is structured as a proactive guide on the steps a user has to take to migrate everything, such as:

1. Update to a Jupyter based install
2. Run `jupyter migrate`
3. Move config files over...
4. etc.

minrk replied:
I think this is a reactive document. Under normal circumstances, users don't have to do anything because migration is automatic. So I expect people coming here to be wondering why it's not already doing what they expect.

Carreau commented on docs/source/migrating.rst:
```diff
@@ -0,0 +1,189 @@
+effect every time you start a new kernel. The less good thing is that
+there isn't a great way to modify the kernelspecs. You can see where the
+file is with ``jupyter kernelspec list``, and then modify
+``kernels/python3/kernel.json`` by hand.
```
You are allowed to auto-promote a2km.

ellisonbg commented on docs/source/migrating.rst:
```diff
@@ -0,0 +1,189 @@
+
+Configuration files
+^^^^^^^^^^^^^^^^^^^
+
```
I would make it clear if the user need to manually copy these or if `jupyter migrate` will

ellisonbg commented on docs/source/migrating.rst:
```diff
@@ -0,0 +1,189 @@
+
+Data files
+^^^^^^^^^^
+
```
Probably describe where the data files used to be, and how/if they need to be moved manually (or by `jupyter migrate`)

Carreau replied:
There is still the issue of custom profile, we might want to expand `jupyter migrate` to take a profile name and a target dir. But probably for another PR.

ellisonbg commented on docs/source/migrating.rst:
```diff
@@ -0,0 +1,189 @@
+
+I want everything
+~~~~~~~~~~~~~~~~~
+
```
I think the migration page should cover the post install steps only as the installation page already has upgrade instructions and users will likely end up there first when running the upgrade.

ellisonbg commented on docs/source/migrating.rst:
```diff
@@ -0,0 +1,189 @@
+is specified, otherwise they will go in a system-wide location (e.g.
+``/usr/local/share/jupyter/nbextensions``). Installation **SHOULD NOT**
+be done by manually guessing where the files should go.
+
```
Also, will nbextensions be automigrated by `jupyter migrate` or does a user have to move them?

minrk replied:
I'll make it clear that everything covered here will be auto-migrated. Under normal circumstances, no users will have to take any actions at all.

ellisonbg replied:
Ahh, got it - it the automigration triggered by running jupyter migrate, or
does that even run itself? This should be clarified.

On Mon, Sep 14, 2015 at 10:05 AM, Min RK <notifications@github.com> wrote:

> In docs/source/migrating.rst
> <https://github.com/jupyter/jupyter/pull/34#discussion_r39418755>:
>
> > +
> > +::
> > +
> > +    ipython install-nbextension [--user] EXTENSION
> > +
> > +They are now installed with
> > +
> > +::
> > +
> > +    jupyter nbextension install [--user] EXTENSION
> > +
> > +They will go in the ``JUPYTER_DATA_DIR`` above if a ``--user`` install
> > +is specified, otherwise they will go in a system-wide location (e.g.
> > +``/usr/local/share/jupyter/nbextensions``). Installation **SHOULD NOT**
> > +be done by manually guessing where the files should go.
> > +
>
> I'll make it clear that everything covered here will be auto-migrated.
> Under normal circumstances, no users will have to take any actions at all.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/jupyter/pull/34/files#r39418755>.
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

Carreau replied:
It is clarified above in the `data files are automatically migrated to their new locations`

ellisonbg commented on docs/source/migrating.rst:
```diff
@@ -0,0 +1,189 @@
+is specified, otherwise they will go in a system-wide location (e.g.
+``/usr/local/share/jupyter/kernels``). Installation **SHOULD NOT** be
+done by manually guessing where the files should go.
+
```
Same comments as above (manual or automatic move).

damianavila commented on docs/source/migrating.rst:
```diff
@@ -0,0 +1,189 @@
+======================
+
+`The Big Split <https://blog.jupyter.org/2015/04/15/the-big-split/>`__
+has changed a few things, moving the various language-agnostict
```
agnostic typo

damianavila commented on docs/source/migrating.rst:
```diff
@@ -0,0 +1,189 @@
+Kernels are installed in much the same way as notebook extensions,
+above:
+
+Notebook extensions used to be installed with
```
notebook extension? copy paste error I think...

Carreau commented:
It's soon gonna be end of working day for @minrk, should we fork and correct there ? Or merge and iterate ?

willingc commented on docs/source/migrating.rst:
```diff
@@ -0,0 +1,189 @@
+
+-  ``IPython.html`` → ``notebook``
+-  ``IPython.html.widgets`` → ``ipywidgets``
+-  ``IPython.kernel`` → ``jupyter_client``, ``ipykernel`` (this became
```
Replace `this` with `IPython.kernel`

minrk commented:
Thanks all! I think I've addressed the review so far.

@ellisonbg the one I'm less clear about how to deal with is your proposed restructuring of the document. What definitely wasn't clear is that most users **should not need to take any action** since all the migrations described in the doc are automatic. The actions that most people may be affected by are:

- updating configuration *after* migration (editing old files won't have an effect)
- things that have actually changed, e.g. `jupyter notebook --profile`

Carreau commented:
Merging and iterating on subsequents PRs.

willingc commented:
@minrk @Carreau @ellisonbg I will review this afternoon and iterate if needed. Looking good.
